src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -204,6 +210,152 @@ private void validate(Schema inputSchema, FailureCollector collector) {
+                                 WindowAggregationConfig.Function function,
+                                 FailureCollector collector) {
+    //Ref :  https://cloud.google.com/bigquery/docs/reference/standard-sql/numbering_functions#ntile
+    if (function.equals(WindowAggregationConfig.Function.N_TILE)) {
+      if (arguments.length != 1) {
+        collector.addFailure(
+          String.format("%s takes exactly 1 argument.", WindowAggregationConfig.Function.N_TILE),
+          "Make sure there is one argument").
+          withConfigProperty(WindowAggregationConfig.NAME_AGGREGATES);
+      } else {
+        try {
+          int percentage = Integer.parseInt(arguments[0].trim());
+          if (percentage < 1) {
+            throw new IllegalArgumentException();
+          }
+        } catch (IllegalArgumentException e) {
+          collector.addFailure(
+            String.format("%s needs an argument of type INTEGER greater than 0",
+                          WindowAggregationConfig.Function.N_TILE), "").
+            withConfigProperty(WindowAggregationConfig.NAME_AGGREGATES);
+        }
+      }
+    }
+
+    // ref : https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#percentile_cont
+    if (function.equals(WindowAggregationConfig.Function.CONTINUOUS_PERCENTILE)
+    || function.equals(WindowAggregationConfig.Function.DISCRETE_PERCENTILE)) {
+      if (arguments.length != 1) {
+        collector.addFailure(
+          String.format("%s takes exactly 1 argument.", function), "Make sure there is one argument").
+          withConfigProperty(WindowAggregationConfig.NAME_AGGREGATES);
+      } else {
+        try {
+          float percentile = Float.parseFloat(arguments[0].trim());
+          if (percentile < 0 || percentile > 1) {
+            throw new IllegalArgumentException();
+          }
+        } catch (IllegalArgumentException e) {
+          collector.addFailure(
+            String.format("%s needs an argument between [0,1]", function), "").
+            withConfigProperty(WindowAggregationConfig.NAME_AGGREGATES);
+        }
+      }
+    }
+
+    // ref : https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#lag
+    if ((function.equals(WindowAggregationConfig.Function.LEAD)
+      || function.equals(WindowAggregationConfig.Function.LAG))) {
+      if (arguments.length != 1) {
+        collector.addFailure(
+          String.format("%s takes only 1 argument.", function), "Make sure there is one argument").
+          withConfigProperty(WindowAggregationConfig.NAME_AGGREGATES);
+      } else {
+        try {
+          int  offset = Integer.parseInt(arguments[0].trim());
+          if (offset < 0) {
+            throw new IllegalArgumentException();
+          }
+        } catch (IllegalArgumentException e) {
+          collector.addFailure(
+            String.format("%s needs an Non Negative Integer argument", function), "").
```

Comment:
nit: "Needs a positive integer argument".

Reply:
Actually it can take ZERo as well. Hence i mentioned Non Negative

Reply:
I see. Only ask would be "non-negative" instead of "Non Negative"